internal/grpc/services/gateway/storageprovider.go

```diff
@@ -179,14 +179,30 @@ func (s *svc) InitiateFileDownload(ctx context.Context, req *provider.InitiateFi
 			}, nil
 		}
 
-		ri, err := s.checkRef(ctx, statRes.Info)
+		ri, protocol, err := s.checkRef(ctx, statRes.Info)
 		if err != nil {
 			log.Err(err).Msg("gateway: error resolving reference")
 			return &gateway.InitiateFileDownloadResponse{
 				Status: status.NewInternal(ctx, err, "error creating container"),
 			}, nil
 		}
 
+		if protocol == "webdav" {
+			// TODO(ishank011): pass this through the datagateway service
+			// for now, we just expose the file server to the user
+			ep, opaque, err := s.webdavRefTransferEndpoint(ctx, statRes.Info.Target, shareChild)
+			if err != nil {
+				return &gateway.InitiateFileDownloadResponse{
+					Status: status.NewInternal(ctx, err, "gateway: error creating container on webdav host: "+p),
+				}, nil
```

Comment:
error message should be: error downloading ....

Reply:
My bad